### PR TITLE
Update use cases in DG

### PR DESCRIPTION
**Pull request by JasonRay168**

### CODE REVIEW

**Review by LordSaumya — APPROVED** [review]
lgtm, needs minor changes

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1045,23 +1045,226 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+  * 1a1. HouR shows an error message.
+
+  Use case returns back to step 1.
+
+* 1b. No employee matches given id.
+  * 1b1. HouR shows an error message.
+
+  Use case returns back to step 1.
+
+* 1c. Dates are of invalid order or format.
```
Comment (inezkok) [inline docs/DeveloperGuide.md:1093]: i think there's another case for when the dates to be added already exist? or if this causes the number of leaves to be over 14 days

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1045,23 +1045,226 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+  Use case returns back to step 1.
+
+* 1c. Remark does not exist in the list of remarks associated with employee with given employee id.
+  * 1c1. HouR shows an error message.
+
+  Use case returns back to step 1.
+
+**Use case: Increasing or decreasing employee overtime hours**
+
+**MSS**
```
Comment (LordSaumya) [inline docs/DeveloperGuide.md:1239]: For MSS should pick either increasing or decreasing, but not both.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1045,23 +1045,226 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    Use case ends.
+
+**Extensions**
+
+* 1a. User leaves out parameters.
+  * 1a1. HouR shows an error message.
+
+  Use case returns back to step 1.
+
+* 1b. No employee matches given employee id.
```
Comment (inezkok) [inline docs/DeveloperGuide.md:1205]: another case would be the given remark already exists

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -1045,23 +1045,226 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+**MSS**
+
+1.  User requests to increase or decrease number of overtime hours taken by employee with given employee id by a given amount.
+2.  HouR updates the number of overtime hours taken by employee with given employee id by the given amount.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. User leaves out parameters.
```
Comment (LordSaumya) [inline docs/DeveloperGuide.md:1248]: Need to be more specific about which parameters are left out

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -1045,23 +1045,226 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+1.  User requests to add a given remark to an employee with a given employee id.
+2.  HouR adds the remark to the list of remarks associated with the employee with the given employee id.
+3.  HouR shows the list of remarks associated with the employee with the given employee id.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. User leaves out parameters.
```
Comment (LordSaumya) [inline docs/DeveloperGuide.md:1200]: Need to be more specific about which parameters are left out

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -1045,23 +1045,226 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+  * 1b1. HouR shows an error message.
+
+  Use case returns back to step 1.
+
+* 1c. Operation given is neither inc (for increase) nor dec (for decrease).
+  * 1c1. HouR shows an error message.
+
+  Use case returns back to step 1.
+
+* 1d. Amount given is not a positive integer.
```
Comment (inezkok) [inline docs/DeveloperGuide.md:1263]: i think there are other cases like when the operation causes the overtime to exceed 72 or be below 0

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -1045,23 +1045,226 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+  * 1b1. HouR shows an error message.
+
+  Use case returns back to step 1.
+
+**Use case: Adding a remark for an employee**
+
+**MSS**
+
+1.  User requests to add a given remark to an employee with a given employee id.
+2.  HouR adds the remark to the list of remarks associated with the employee with the given employee id.
```
Comment (LordSaumya) [inline docs/DeveloperGuide.md:1193]: The use case stories should alternate between the system and the user

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -1045,23 +1045,226 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+**MSS**
+
+1.  User requests to edit a given allocated leave date for an employee with a given id to the given new leave date.
+2.  HouR edits the given allocated leave date to the given new leave date for the employee with given id.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. User leaves out parameters.
```
Comment (LordSaumya) [inline docs/DeveloperGuide.md:1142]: Need to be more specific about which parameters

**Review by inezkok — COMMENTED** [review]
lgtm! i commented on some minor things you might have missed out on!

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -1045,23 +1045,226 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+  Use case returns back to step 1.
+
+**Use case: Deleting a remark for an employee**
+
+**MSS**
+
+1.  User requests to delete a given remark from an employee with a given employee id.
+2.  HouR deletes the remark from the list of remarks associated with the employee with the given employee id.
+3.  HouR shows the list of remarks associated with the employee with the given employee id.
```
Comment (LordSaumya) [inline docs/DeveloperGuide.md:1216]: MSS should alternate between user and system

**Review by inezkok — APPROVED** [review]
lgtm!

**Pull request merged**